crates/extensions/c8y_mapper_ext/src/tests.rs

```diff
@@ -2198,6 +2540,244 @@ async fn handle_log_upload_successful_cmd_for_child_device() {
     .await;
 }
 
+#[tokio::test]
+async fn c8y_mapper_nested_child_alarm_mapping_to_smartrest() {
+    let cfg_dir = TempTedgeDir::new();
+    let (mqtt, _http, _fs, mut timer, _dl) = spawn_c8y_mapper_actor(&cfg_dir, true).await;
+
+    timer.send(Timeout::new(())).await.unwrap(); //Complete sync phase so that alarm mapping starts
+    let mut mqtt = mqtt.with_timeout(TEST_TIMEOUT_MS);
+    skip_init_messages(&mut mqtt).await;
+
+    let reg_message = MqttMessage::new(
+        &Topic::new_unchecked("te/device/immediate_child//"),
+        json!({
+            "@type":"child-device",
+            "@parent":"device/main//",
+            "@id":"immediate_child"
+        })
+        .to_string(),
+    );
+    mqtt.send(reg_message).await.unwrap();
+
+    let reg_message = MqttMessage::new(
+        &Topic::new_unchecked("te/device/nested_child//"),
+        json!({
+            "@type":"child-device",
+            "@parent":"device/immediate_child//",
+            "@id":"nested_child"
+        })
+        .to_string(),
+    );
+    mqtt.send(reg_message).await.unwrap();
+
+    mqtt.send(MqttMessage::new(
+        &Topic::new_unchecked("te/device/nested_child///a/"),
+        json!({ "severity": "minor", "text": "Temperature high","time":"2023-10-13T15:00:07.172674353Z" }).to_string(),
+    ))
+    .await
+    .unwrap();
+
+    mqtt.skip(2).await;
```

Comment:
But the order of messages is fixed for any given operation. So, I feel that skipping is OK here instead of just asserting the topics.

Reply:
> If we do not skip, the assert happens on the wrong topic and tests will fail

This is exactly what we're trying to prevent. When you're skipping 2 messages here, you're not skipping any two messages. You're specifically skipping 2 messages that are probably not relevant for the test. But, still it is important to be sure that those 2 messages are exactly what you expected. Here, for example, you expect to skip the first 2 registration messages. What if one of them failed and sent an error message to `tedge/errors` instead? It'd fail in the next assertion but with some really weird errors than what you expected. So, that's why I'm suggesting a loose assertion on `("c8y/s/us", "101")` topic.

Reply:
ok, updated the tests